src/prefect/runner/storage.py

```diff
@@ -156,8 +162,12 @@ def _repository_url_with_credentials(self) -> str:
             if isinstance(self._credentials, Block)
             else deepcopy(self._credentials)
         )
-        if isinstance(credentials.get("access_token"), Secret):
-            credentials["access_token"] = credentials["access_token"].get()
+
+        for k, v in credentials.items():
+            if isinstance(v, Secret):
+                credentials[k] = v.get()
+            elif isinstance(v, SecretStr):
+                credentials[k] = v.get_secret_value()
```

Comment:
if someone passes `credentials={"foo_token": GitlabCredentials.load(...).access_token}` would that break?

i.e. does the key name matter for us?

Reply:
The key name does matter, but an invalid key will raise a `ValueError` in `_format_token_from_credentials`, so the key name doesn't need to be validated here.